### PR TITLE
Make replays pause with middle mouse button instead of exiting

### PR DESCRIPTION
**Pull request by Joehuu**
- Addresses code comment added in https://github.com/ppy/osu/pull/9540

### CODE REVIEW

File: osu.Game.Tests/Visual/Gameplay/TestSceneReplayPlayer.cs [hunk 1]
```diff
@@ -23,14 +24,29 @@ public override void SetUpSteps()
             AddUntilStep("player loaded", () => Player.IsLoaded);
         }
 
-        [Test]
-        public void TestPause()
+        [TestCase("space")]
+        [TestCase("mouse middle")]
+        public void TestPause(string action)
```
Comment (peppy) [inline osu.Game.Tests/Visual/Gameplay/TestSceneReplayPlayer.cs:29]: Just make two test scenes at this point. Using a string like this is a bit weird.

File: osu.Game/Screens/Play/ReplayPlayer.cs [hunk 2]
```diff
@@ -81,6 +81,7 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     return true;
 
                 case GlobalAction.TogglePauseReplay:
+                case GlobalAction.PauseGameplay:
```
Comment (peppy) [inline osu.Game/Screens/Play/ReplayPlayer.cs:84]: I'm not sure this makes sense.

**Comment by peppy** [comment]
Ignoring review feedback, I'm not sure this makes sense. I'd probably expect middle mouse did nothing than paused in such a case.

Interested in more feedback.

**Comment by Joehuu** [comment]
I was just following what you said in the mentioned PR:

> Note that for the moment this just triggers standard "quit" behaviour (for cases like replays). This can change in the future once we have the ability to pause replays as well, but having the binding in place earlier is favourable IMO.

**Comment by bdach** [comment]
Yeah I tend to agree that this doesn't need to be a thing. It feels really weird to alias the two actions because of how differently they work (gameplay pause requires hold, and results in a very modal interaction, i.e. a pause overlay you can't get rid of, but replay pause is just an instantaneous toggle).

If anyone _really_ wants this for whatever reason then there is nothing stopping them from aliasing both actions to the same key in settings. I think this is a "remove comment and continue on our merry way" situation.

**Comment by Joehuu** [comment]
> I'd probably expect middle mouse did nothing than paused in such a case.

I see doing nothing as a downgrade to `master` and this PR because there are mouse buttons not being used. You can't even pause with just mouse by default.

> It feels really weird to alias the two actions because of how differently they work

I guess the solution for this is just to also add `M3` to pause / resume replay alongside space?

> gameplay pause requires hold

Not necessarily for the first one as 0 delay is an option.

> and results in a very modal interaction, i.e. a pause overlay you can't get rid of, but replay pause is just an instantaneous toggle

Probably an oversight when `PauseGameplay` is added in https://github.com/ppy/osu/pull/9540 to match stable. It should behave like `Back`.

> If anyone really wants this for whatever reason then there is nothing stopping them from aliasing both actions to the same key in settings. I think this is a "remove comment and continue on our merry way" situation.

This isn't possible with current `master` with these settings and would still need my `ReplayLoaded` changes:
image

**Comment by peppy** [comment]
I think I'm okay with this now. I've moved replay key bindings to their own section, as I am 100% sure there will be more in the future, and it makes it a lot easier to visually understand (one of my main concerns with this PR was having two "pause" bindings next to each other in the settings).

**Review by peppy — APPROVED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**